### DISCUSSION TITLE
normalize_names prepending underscore _ on SQL keywords but...

### DISCUSSION THREAD

**Discussion by dmcmurchy** [duckdb/duckdb#12795] [General]
I understand that the `read_csv` [normalize_names](https://duckdb.org/docs/data/csv/overview#parameters) functionality was [designed ](https://github.com/duckdb/duckdb/pull/1875) to prepend the `_` on SQL keywords, but the regular import without normalize doesn't perform this transformation. Additionally, there doesn't appear to be anything to stop someone from creating a table using keywords as column names. Prepending of the underscore to column names that match keywords when using `normalize_names` seems like an unnecessary transformation of the data on an otherwise great feature and I think this should be removed.

![image](https://github.com/duckdb/duckdb/assets/5430744/d6f3bb50-55fa-4710-9070-b7d0b4e00630)

![image](https://github.com/duckdb/duckdb/assets/5430744/489aea33-c41e-479a-9197-b250874072ca)

![image](https://github.com/duckdb/duckdb/assets/5430744/5b0bf444-f237-4a0e-a568-ccda21acea4f)

![image](https://github.com/duckdb/duckdb/assets/5430744/cf9e8289-ff34-411e-bb9b-6b533d9fc515)

**Comment by Tishj** [comment]
It looks like this is following a spec:
<https://github.com/duckdb/duckdb/pull/1875>

**Comment by dmcmurchy** [comment]
Yes, I included the link in my original message. The discussion [prior](https://github.com/duckdb/duckdb/pull/1789) to that one doesn't mention the need to sanitize the column names based on keywords.  The point was, why does it bother doing a check against the keywords when nothing else appears to perform a sanity check on the column names and continues to function? 

There are currently **479** words that if they appear as a column name in a CSV file will have underscore prepended to the name requiring the user to go and rename it after the fact or come up with another way to deal the column names. 

The user documentation for `normalize_names` states: `Boolean value that specifies whether or not column names should be normalized, removing any non-alphanumeric characters from them.` There is no mention that it will perform a check against keywords and prepend underscores.
